### DISCUSSION TITLE
Is the summary table supposed to be part of the default configuration?

### DISCUSSION THREAD

**Discussion by Plasmadog** [dotnet/BenchmarkDotNet#2354] [Q&A]
Everything I see about BenchmarkDotNet, including the Getting Started page, and other online tutorials, all seem indicate that simply running a benchmark with the default configuration should output a summary table to the console. That is not the result I get. I do get tables as markdown and CSV in the results folder, but in the console output I see only the log messages for each separate benchmarked method.
Is that expected? Should the table be shown in the console, or am I misreading the guides?

**Comment by adamsitnik** [comment]
Hi @Plasmadog 

How exactly do you run your benchmarks? Do you customize anything?

**Reply by Plasmadog** [reply]
No customizations, just following the Getting Started guide exactly. Whether I run via Visual Studio's F5 or via the command line (dotnet run -c Release), I get the same result.

**Reply by adamsitnik** [reply]
Could you please share your exact code and log output via gist?

**Reply by Plasmadog** [reply]
Sure:
[https://gist.github.com/Plasmadog/048d8c839f7f96130b52cf1ef757bf33](https://gist.github.com/Plasmadog/048d8c839f7f96130b52cf1ef757bf33)

**Reply by AndreyAkinshin** [reply]
@Plasmadog it seems that BenchmarkDotNet hanged just before printing the summary table. Could you please attach to the running process using your IDE and check out the stack trace?

**Reply by Plasmadog** [reply]
I'm not having any success with that. For a start, I'm not sure which process to attach to. If each benchmark is run in a different process, what process is responsible for the final summary? Nothing I have tried has triggered any sort of breakpoint in the IDE. Then again, I'm not sure what to expect from a Release build.

It does appear that nothing after the BenchmarkRunner.Run<Md5VsSha256>(); line is executed.

**Reply by timcassell** [reply]
@Plasmadog You can attach to the host process to see where it hangs. If you want to attach to the benchmark process, you can use the `DebugBuildConfig`. See the [troubleshooting guide](https://andreyakinshin.github.io/bdn-docs-wip/articles/guides/troubleshooting.html).

**Reply by Plasmadog** [reply]
Thanks, but I have not found the troubleshooting guide to be helpful. For a start, it does not say anything about _how_ to use DebugBuildConfig, only that it _can_ be used. I did try DebugInProcessConfig using the example provided, but that actually did show the summary and did not throw any exception. I tried the same code with DebugBuildConfig in place of DebugInProcessConfig, on the assumption that that is how it is intended to be used, but that just led to a lot of error messages related to locked files. So, if you want me to do any of those things, some specific step-by-step instructions would be helpful.

Regarding attaching to the "host process", I'm not sure which one that is. Is it the console app itself? The one named after my project? I have tried that, and the debugger does not stop at any point. I have tried both of the processes that have "--benchmarkId $someNumber --benchmarkName $theName" as parameters, and they also do not stop.

**Reply by Plasmadog** [reply]
Update - I did get the DebugBuildConfig to run, but that also shows the summary without failure.

**Reply by Plasmadog** [reply]
OK, so I have found where the problem is. I still haven't got a stack trace, but the issue is with the HTML exporter, which is enabled by default.
If I run this:
```
var summary = BenchmarkRunner.Run<Md5VsSha256>(
                ManualConfig.
                    CreateMinimumViable()
                    .AddExporter(CsvExporter.Default)
                    .AddExporter(MarkdownExporter.Default)
                    .AddExporter(HtmlExporter.Default));
```

...I get no summary. However, if I leave the HtmlExporter out of that list it works OK.

**Reply by timcassell** [reply]
> Regarding attaching to the "host process", I'm not sure which one that is. Is it the console app itself?

Yes.

> the debugger does not stop at any point

That indicates an infinite loop, or a deadlock. Once you have the debugger attached, you should be able to manually pause it and find where that infinite loop/deadlock is occurring.

> the issue is with the HTML exporter

Interesting. I don't know anything about that code, but nice job narrowing it down.
